### PR TITLE
Deprecate expr_form argument in the LocalClient

### PR DESCRIPTION
**Pull request by terminalmage**
The published commands and the events all use tgt_type, expr_form is
only used in the LocalClient, which has been confusing to many users.
This puts expr_form on a deprecation path, allowing it to continue to be
supported while tgt_type becomes the new official argument name.

### CODE REVIEW

File: doc/topics/event/master_events.rst [hunk 1]
```diff
@@ -133,8 +133,10 @@ Runner Events
     :var tgt: The target expression (e.g. ``*``). Included for ``state`` and
         ``function`` types only.
     :var args: The args passed to the orchestration job. **Note:** for
-        ``state`` and ``function`` types, also includes an ``expr_form`` which
-        shows what kind of match (``glob``, ``pcre``, etc.) was used.
+        ``state`` and ``function`` types, also includes a ``tgt_type`` value
+        which shows what kind of match (``glob``, ``pcre``, etc.) was used.
+        This value was named ``expr_form`` in the 2016.11 release cycle but has
+        been renamed to ``tgt_type`` in Nitrogen for consistency with other.
```
Comment (rallytime) [inline doc/topics/event/master_events.rst]: For consistency with other?

**Review by rallytime — COMMENTED** [review]
I only had one small comment after the first run through here. This is a thorough change - nice work!

File: doc/topics/event/master_events.rst (hunk 1 above)
Comment (terminalmage) [inline doc/topics/event/master_events.rst]: Derp. Thanks, fixing.

Comment (terminalmage) [inline doc/topics/event/master_events.rst]: Fixed in 26600bc.

**Comment by cachedout** [comment]
I think we should probably make a note in the Nitrogen release notes before we merge this so that we don't forget down the line.

**Comment by terminalmage** [comment]
Yeah good call, I had intended to do this but forgot.

**Comment by terminalmage** [comment]
OK, notice added to release notes.

**Comment by cachedout** [comment]
@terminalmage This triggered a test failure: https://jenkins.saltstack.com/job/PR/job/salt-pr-rs-cent7-n/7318/testReport/junit/unit.client_test/LocalClientTestCase/test_cmd_subset/

**Comment by terminalmage** [comment]
Yeah, didn't check tests for expr_form usage. Just pushed an update.

**Pull request merged**